java/flight/flight-core/src/main/java/org/apache/arrow/flight/SessionOptionValueFactory.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.flight;
+
+import org.apache.arrow.flight.SessionOptionValue;
+
+public class SessionOptionValueFactory {
+    public static SessionOptionValue makeSessionOption(String value) {
+        return new SessionOptionValueString(value);
+    }
+
+    public static SessionOptionValue makeSessionOption(bool value) {
+        return new SessionOptionValueBool(value);
+    }
+
+    public static SessionOptionValue makeSessionOption(int value) {
+        return new SessionOptionValueInt(value);
+    }
+
+    public static SessionOptionValue makeSessionOption(long value) {
+        return new SessionOptionValueLong(value);
+    }
+
+    public static SessionOptionValue makeSessionOption(float value) {
+        return new SessionOptionValueFloat(value);
+    }
+
+    public static SessionOptionValue makeSessionOption(double value) {
+        return new SessionOptionValueDouble(value);
+    }
+
+    public static SessionOptionValue makeSessionOption(String[] value) {
+        return new SessionOptionValueStringList(value);
+    }
+}
+
+class SessionOptionValueString {
```

Comment:
Are options nullable? Is there any way for a user to "unset" an option? For example they might want to do this to revert to a server-side default.

Reply:
Good point.  Will consider how to integrate that; I believe the Proto default is something like the first value, empty, so an empty string?  But that could conflict with other intents, so we might need an additional dummy oneof state.

Reply:
I -think-, unless there are broken edge cases for this in Proto, the most straightforward thing here would be to handle a value variant that maps to an unset `oneof` over the wire.  (a) I think we can add that later though and (b) does that sound sane (both the impl and feasibility of deferring it for now)?

Reply:
> I -think-, unless there are broken edge cases for this in Proto, the most straightforward thing here would be to handle a value variant that maps to an unset `oneof` over the wire. (a) I think we can add that later though and (b) does that sound sane (both the impl and feasibility of deferring it for now)?

@lidavidm any objections to this idea before it moves ahead, "unset" options using the VALUE_NOT_SET case over the wire and presumably then std::variant::valueless_by_exception() and having a visit(Void) overload in Javaland?

Reply:
Implemented this in both languages.

Reply:
+Added comment to Proto (already in C++ and Java types/visitors respectively)